apps/core/app/[locale]/(default)/account/[tab]/_components/addresses-list.tsx

```diff
@@ -1,36 +1,64 @@
+'use client';
+
 import { Button } from '@bigcommerce/components/button';
 import { useTranslations } from 'next-intl';
+import { Dispatch, SetStateAction, useState } from 'react';
 
 import { getCustomerAddresses } from '~/client/queries/get-customer-addresses';
 import { Link } from '~/components/link';
 
-type Addresses = NonNullable<Awaited<ReturnType<typeof getCustomerAddresses>>>['addresses'];
+import { deleteAddress } from '../_actions/delete-address';
 
-interface Props {
-  customerAddressBook: Addresses;
-}
+import { Modal } from './modal';
+
+type Addresses = NonNullable<Awaited<ReturnType<typeof getCustomerAddresses>>>['addresses'];
 
-const AddressChangeButtons = () => {
+const AddressChangeButtons = ({
+  addressId,
+  setter,
+}: {
+  addressId: number;
+  setter: Dispatch<SetStateAction<Addresses>>;
```

Comment:
Could we make this API a little more generic instead of consuming a react-specific API? e.g.
```ts
interface Props {
  onDelete(address: Addresses): void
}
```

Reply:
changed